include/aabbox3d.h

```diff
@@ -15,20 +24,33 @@ namespace core
 
 //! Axis aligned bounding box in 3d dimensional space.
 /** Has some useful methods used with occlusion culling or clipping.
+ * @TODO UVector was introduced to allow for usage other than of vector3d<float> (i.e. vectorSIMDf)
+ *   and defaulted to allow for gradual migration of vector3d-based usage/implementation of aabbox3df
+ *   to vectorSIMDf (i.e. vector3df_SIMD) everywhere in the codebase
 */
-template <class T>
+template<class T, class UVector>

```

Comment:
these modifications are not needed, `aabbox3d` is a legacy class I intend to get rid of in the near future

Reply:
`aabbox3d` is used everywhere in the codebase. If this is a legacy which is used everywhere else, then don't you think it should be gradually upgraded than just ignoring it? its methods are possibly wrong (e.g. getEdges) but then they can simply be rectified. It still plays an important role throughout the entire system's structure.